stack/tests/deployment_test.yaml

```diff
@@ -157,6 +157,36 @@ tests:
       - equal:
           path: spec.template.spec.initContainers[0].image
           value: "node:20.0.0"
+  - it: uses supplied sidecar image string value
+    set:
+      services:
+        service1:
+          sidecars:
+            - name: sidecar1
+              image: "alpine:latest"
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].image
+          value: "nginx:latest"
+      - equal:
+          path: spec.template.spec.containers[1].image
+          value: "alpine:latest"
+  - it: formats supplied sidecar image tag and repo
+    set:
+      services:
+        service1:
+          sidecars:
+            - name: sidecar2
+              image:
+                repository: "node"
+                tag: "20.0.0"
```

Comment:
can you update the type annotations in the values.yml too?

Reply:
they update automatically but the only way to get schema changes for this is if we make the sidecars list non-empty by default 😭